lib/rules/rule-empty-line-before/README.md

```diff
@@ -0,0 +1,223 @@
+# rule-empty-line-before
+
+Require or disallow an empty line before rules.
+
+```css
+a {}
+      /* ← */
+b {}  /* ↑ */
+/**      ↑
+ * This line */
+```
+
+If the rule is the very first node in a stylesheet then it is ignored.
+
+## Options
+
+`string`: `"always"|"never"|"always-multi-line"|"never-multi-line"`
+
+### `"always"`
+
+There *must always* be an empty line before rules.
+
+The following patterns are considered warnings:
+
+```css
+a {} b {}
+```
+
+```css
+a {}
+b {}
+```
+
+The following patterns are *not* considered warnings:
+
+```css
+a {}
+
+b {}
+```
+
+### `"never"`
+
+There *must never* be an empty line before rules.
+
+The following patterns are considered warnings:
+
+```css
+a {}
+
+b {}
+```
+
+The following patterns are *not* considered warnings:
+
+```css
+a {} b {}
+```
+
+```css
+a {}
+b {}
+```
+
+### `"always-multi-line"`
+
+There *must always* be an empty line before multi-line rules.
+
+The following patterns are considered warnings:
+
+```css
+a
+{}
+b
+{}
+```
+
+The following patterns are *not* considered warnings:
+
+```css
+a
+{}
+
+b
+{}
+```
+
+### `"never-multi-line"`
+
+There *must never* be an empty line before multi-line rules.
+
+The following patterns are considered warnings:
+
+```css
+a
+{}
+
+b
+{}
+```
+
+The following patterns are *not* considered warnings:
+
+```css
+a
+{}
+b
+{}
+```
+
+## Optional secondary options
+
+### `except: ["after-single-line-comment", "inside-block-and-after-rule", "first-nested"]`
+
+#### `"after-single-line-comment"`
```

Comment:
This rule mimics the options from the previous two rules. Only the `rule-nested...` had the after single line comment exception, as there was a specific request for that functionality.